config/config.go

```diff
@@ -408,6 +433,48 @@ func (c *Config) InitializeWarpQuorums() error {
 	return nil
 }
 
+func (p *PChainAPI) Validate() error {
+	if _, err := url.ParseRequestURI(p.BaseURL); err != nil {
+		return fmt.Errorf("invalid base URL: %w", err)
+	}
+	p.options = make([]rpc.Option, 0, len(p.QueryParams))
+	for key, value := range p.QueryParams {
+		p.options = append(p.options, rpc.WithQueryParam(key, value))
```

Comment:
Option creation could be moved into client implementation/usage as it is the one aware of rpc usage.

For example in `validators/canonical_validator_client.go` you create a config struct being a subset of the global one
```
type Config struct {
  QueryParams map[string]string
  HttpHeaders map[string]string
}
```
This Config object becomes an argument to the `NewCanonicalValidatorClient` factory. Then the factory is responsible to create the Options and store it the same way you did.

Reply:
Totally can see that being an option, and thanks for the suggestion. I don't anticipate there being too many differences in the usage of query parameters throughout the relayer, so prefer to leave a unified format when parsing the config under `PChainAPI` and `InfoAPI`.